service/awsconfig/v5/resource/loadbalancer/current.go

```diff
@@ -2,9 +2,85 @@ package loadbalancer
 
 import (
 	"context"
+
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/awsconfig/v5/key"
+)
+
+const (
+	cloudProviderClusterTagValue = "owned"
+	cloudProviderServiceTagKey   = "kubernetes.io/service-name"
 )
 
-// TODO
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	currentState := LoadBalancerState{}
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return currentState, microerror.Mask(err)
```

Comment:
It would be nice to be explicit here. I fear somebody like me just mutates current state or refactors the method and then has to deal with this which is just not necessary. In golang is a saying like "be explicit, not smart". I try to live that. I see that this also applies to other methods. We could also just work with pointers for our state dispatching to be able to simply return `nil` then we would not have this problem at all.

Reply:
Ack, better to be explicit here. I've changed to use pointers for the state.